### PR TITLE
build: reorg Makefile

### PR DESCRIPTION
**Pull request by disksing**
Signed-off-by: disksing <i@disksing.com>

### What problem does this PR solve?
Makefile is not well organized and there many stale targets.

### What is changed and how it works?
- remove some targets
- reorder some targets
- merge some targets
- rewrite some targets
- remove go version check and GO111MODULE setting

### Check List

Tests
- Unit test
### Release note

```release-note
None
```

### CODE REVIEW

**Review by JmPotato — APPROVED** [review]

File: plugin/scheduler_example/Makefile [hunk 1]
```diff
@@ -1,5 +1,5 @@
 evictLeaderPlugin.so: *.go
-	GO111MODULE=on GO_ENABLED=1 go build  -buildmode=plugin -o evictLeaderPlugin.so *.go
+	CGO_ENABLED=1 go build -buildmode=plugin -o evictLeaderPlugin.so *.go
```
[the remark this answers is not in the document]
Comment (disksing) [reply plugin/scheduler_example/Makefile:2]: the default value is `on` after go1.16

File: Makefile [hunk 2]
```diff
@@ -107,160 +69,196 @@ ifneq ($(DASHBOARD_DISTRIBUTION_DIR),)
...
 	for PKG in $(TEST_PKGS); do\
 		set -euo pipefail;\
-		CGO_ENABLED=1 GO111MODULE=on go test -race -covermode=atomic -coverprofile=coverage.tmp -coverpkg=./... $$PKG 2>&1 | grep -v "no packages being tested" && tail -n +2 coverage.tmp >> covprofile || { $(FAILPOINT_DISABLE); rm coverage.tmp && exit 1;}; \
+		CGO_ENABLED=1 go test -race -covermode=atomic -coverprofile=coverage.tmp -coverpkg=./... $$PKG 2>&1 | grep -v "no packages being tested" && tail -n +2 coverage.tmp >> covprofile || { $(FAILPOINT_DISABLE); rm coverage.tmp && exit 1;}; \
 		rm coverage.tmp;\
 	done
 	@$(FAILPOINT_DISABLE)
 
-# The command should be used in daily CI，it will split some tasks to run parallel.
-# It should retain report.xml,coverage,coverage.xml and package.list to analyze.
```
Comment (bufferflies) [inline Makefile:191]: why remove it，it will run in ci-pipeline.

Comment (disksing) [reply Makefile:191]: I didn't find anywhere using it. I think now coverage is reported by `test-with-cover`. You can see this PR is normally generating coverage report.

Comment (disksing) [reply Makefile:191]: add back.

**Review by rleungx — APPROVED** [review]

**Pull request merged**